File: backend/StudentBlogAPI/Model/DTOs/CommenDtos.cs
using System.ComponentModel.DataAnnotations;

namespace StudentBlogAPI.Model.DTOs;

public record CommentResDto(
    int Id,
    int PostId,
    int UserId,
    string Content,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public record CommentInput(
    [MinLength(6)] [MaxLength(140)] string Content
    );

public record CreateCommentDto(
    int PostId,
    int UserId,
    string Content
);

public record UpdateCommentDto(
    int Id,
    int UserId,
    string Content
);

File: frontend/dtos/CommenDtos.ts
type CommentResDto = {
  id: number;
  postId: number;
  userId: number;
  content: string;
  createdAt: string; // Represents a DateTime. Format: ISO string
  updatedAt: string; // Represents a DateTime. Format: ISO string
}
type CommentInput = {
  content: string;
}
type CreateCommentDto = {
  postId: number;
  userId: number;
  content: string;
}
type UpdateCommentDto = {
  id: number;
  userId: number;
  content: string;
}
